### PR TITLE
Refs #30423 - preserve old pulpcore MEDIA_ROOT on upgrades

### PR DESCRIPTION
**Pull request by wbclark**
Requires: https://github.com/theforeman/puppet-pulpcore/pull/113
Requires: https://github.com/theforeman/puppet-foreman_proxy_content/pull/275

### CODE REVIEW

File: config/foreman-proxy-content-answers.yaml [hunk 1]
```diff
@@ -14,6 +14,7 @@ certs:
 foreman_proxy_content:
   proxy_pulp_isos_to_pulpcore: false
   proxy_pulp_yum_to_pulpcore: false
+  pulpcore_media_root: /var/lib/pulp
```
Comment (ekohl) [inline config/foreman-proxy-content-answers.yaml:17]: This matches the default, so no need to include this.

**Review by ekohl — CHANGES_REQUESTED** [review]
I don't think this is right. IMHO we should get everyone on the same layout and move files around if needed. The problem is that this means every upgrading user will forever be on a different layout than fresh installs which is hard to support.

Let's first agree on the correct layout (https://github.com/theforeman/puppet-pulpcore/pull/113#pullrequestreview-450769300) and then think about how to deal with upgrades.

File: config/foreman-proxy-content-answers.yaml (hunk 1 above)
Comment (wbclark) [reply config/foreman-proxy-content-answers.yaml:17]: It's included to make it easy to tell new installs apart from upgrades.

Comment (wbclark) [reply config/foreman-proxy-content-answers.yaml:17]: https://github.com/theforeman/foreman-installer/pull/543/files#diff-77cd90f81394cb4c12f94334eb5e1489R9

**Comment by wbclark** [comment]
> I don't think this is right. IMHO we should get everyone on the same layout and move files around if needed.

Is there anything stopping us from doing this within a migration @ekohl ?

**Comment by ekohl** [comment]
Yes, it's not the appropriate place. Migrations are only for the configs and run in a RPM `%post` hook. Moving the files there is guaranteed to break things.

**Comment by wbclark** [comment]
> Yes, it's not the appropriate place. Migrations are only for the configs and run in a RPM `%post` hook. Moving the files there is guaranteed to break things.

How then do you propose moving the uploaded content to the new MEDIA_ROOT directory on upgrades? I think you must have something in mind other than asking users to do it manually?

**Comment by ekohl** [comment]
> How then do you propose moving the uploaded content to the new MEDIA_ROOT directory on upgrades? I think you must have something in mind other than asking users to do it manually?

A hook in `pre` makes the most sense to me. However, the steps we need to take (and we're extremely late in doing this):
* Agree on a directory layout
* Get the proper SELinux policy
* Work out a migration

This should really have been done before we even started to ship it in nightly. Now I just wrote something in puppet-pulpcore but forgot a TODO so others didn't see it.

**Comment by ekohl** [comment]
Closing in favor of the alternative approach.

**Pull request closed without merging**